### PR TITLE
[Merged by Bors] - feat: port Algebra.Algebra.Hom

### PR DESCRIPTION
**Pull request by Komyyy**
---

- [x] depends on: #2374
- [x] depends on: #2385

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Komyyy** [comment]
I created the PR #2374 to port this file.

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2374
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2374~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2374~~
* ~~leanprover-community/mathlib4#2385~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Algebra/Algebra/Hom.lean [hunk 1]
```diff
@@ -103,35 +127,43 @@ theorem toFun_eq_coe (f : A →ₐ[R] B) : f.toFun = f :=
...
-  commutes f := f.commutes'
+attribute [coe] AlgHom.toRingHom
 
-instance coeRingHom : Coe (A →ₐ[R] B) (A →+* B) :=
+instance coeOutRingHom : CoeOut (A →ₐ[R] B) (A →+* B) :=
   ⟨AlgHom.toRingHom⟩
-#align alg_hom.coe_ring_hom AlgHom.coeRingHom
+#align alg_hom.coe_ring_hom AlgHom.coeOutRingHom
+
+-- Porting note: The new definition for an notaion `↑`.
```
Comment (jcommelin) [inline Mathlib/Algebra/Algebra/Hom.lean:136]:
```suggestion
-- Porting note: A new definition underlying a coercion `↑`.
```

File: Mathlib/Algebra/Algebra/Hom.lean [hunk 2]
```diff
@@ -103,35 +127,43 @@ theorem toFun_eq_coe (f : A →ₐ[R] B) : f.toFun = f :=
...
-  ⟨fun f => ↑(f : A →+* B)⟩
-#align alg_hom.coe_monoid_hom AlgHom.coeMonoidHom
+instance coeOutMonoidHom : CoeOut (A →ₐ[R] B) (A →* B) :=
+  ⟨AlgHom.toMonoidHom'⟩
+#align alg_hom.coe_monoid_hom AlgHom.coeOutMonoidHom
 
-instance coeAddMonoidHom : Coe (A →ₐ[R] B) (A →+ B) :=
-  ⟨fun f => ↑(f : A →+* B)⟩
-#align alg_hom.coe_add_monoid_hom AlgHom.coeAddMonoidHom
+-- Porting note: The new definition for an notaion `↑`.
```
Comment (jcommelin) [inline Mathlib/Algebra/Algebra/Hom.lean:144]:
```suggestion
-- Porting note: A new definition underlying a coercion `↑`.
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**